src/NaluCommNeighbors.C

```diff
@@ -84,7 +85,17 @@ CommNeighbors::CommNeighbors( stk::ParallelMachine comm, const std::vector<int>&
 {
   m_send.resize(m_size);
   m_recv.resize(m_size);
-  std::sort(m_neighbor_procs.begin(), m_neighbor_procs.end());
+  stk::util::sort_and_unique(m_send_procs);
+  stk::util::sort_and_unique(m_recv_procs);
+  MPI_Info info;
+  MPI_Info_create(&info);
+  int reorder = 0;
+  const int* weights = (int*)MPI_UNWEIGHTED;
+  MPI_Dist_graph_create_adjacent(comm,
```

Comment:
It might still make sense to wrap this function with an interface, so you're not committed to it.  I haven't benchmarked MPI neighborhood collectives.  They _should_ be optimized.

Reply:
Yes some extra wrapping might be a good idea here. I was worried about performance, but I ran these changes on the 8192-rank case on the cori machine a few times over the weekend and performance seems to be ok. But of course that doesn't speak for all MPI implementations on all platforms...

Reply:
@alanw0 If you do wrap this, please consider an interface that takes offsets rather than (or in addition to) counts.  That would give implementations the option to thread-parallelize underneath.